src/datumaro/experimental/categories.py

```diff
@@ -45,7 +46,7 @@ def from_str(cls, text: str) -> GroupType:
             raise ValueError(f"Invalid GroupType: {text}")
 
 
-@dataclass(frozen=True)
+@dataclass()
```

Comment:
Are you sure this change is in line with the design? I think in general it is saver to keep the data immutable and make a new object explicitly whenever we need to change the data.